app/src/main/java/com/imashnake/animite/features/navigationbar/NavigationBar.kt

```diff
@@ -52,12 +53,17 @@ fun NavigationBar(
         ) { WindowInsets.displayCutout } else { WindowInsets(0.dp) }
     ) {
         val currentDestination by navController.currentDestinationAsState()
-
-        NavigationBarPaths.values().forEach { destination ->
+        NavigationBarPaths.entries.forEach { destination ->
+            val isCurrentDestOnBackStack = navController.isRouteOnBackStack(destination.route)
             NavigationBarItem(
                 modifier = Modifier.navigationBarsPadding(),
                 selected = currentDestination?.startDestination == destination.route,
                 onClick = {
+                    if (isCurrentDestOnBackStack) {
```

Comment:
What's this even supposed to fix

Reply:
See commit description: https://github.com/raamcosta/compose-destinations/blob/46099dc0f6066a1fb343b1f107445bfa29cbca15/sample/src/main/java/com/ramcosta/destinations/sample/ui/composables/BottomBar.kt#L33-L58